modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockRelations.scala

```diff
@@ -0,0 +1,32 @@
+package org.tessellation.dag.l1.domain.block
+
+import cats.effect.kernel.Async
+import cats.syntax.eq._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.{BlockReference, transaction}
+import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
+
+object BlockRelations {
+  def areRelated[F[_]: Async: KryoSerializer](block: Hashed[DAGBlock]): F[Signed[DAGBlock] => Boolean] =
```

Comment:
imo it should return a function: Signed[DAGBlock] => F[Boolean]. Why you need to lift the whole fn to effect?

Reply:
Comment below.